### PR TITLE
Support parsing atom properties and atom values from cxsmiles

### PR DESCRIPTION
**Pull request by greglandrum**
There's a bit of additional refactoring of the cxsmiles parsing code too.

### CODE REVIEW

File: Code/GraphMol/SmilesParse/CXSmilesOps.cpp [hunk 1]
```diff
@@ -25,16 +25,57 @@ namespace parser {
...
+  first += 5;
+  unsigned int atIdx = 0;
+  while (first != last && *first != '$') {
+    std::string tkn = read_text_to(first, last, ';', '$');
+    if (tkn != "") {
+      mol.getAtomWithIdx(atIdx)->setProp(RDKit::common_properties::molFileValue,
+                                         tkn);
+    }
+    ++atIdx;
+    if (first != last && *first != '$') ++first;
```
Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/CXSmilesOps.cpp]: first cannot be last here due to the while loop condition.

Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/CXSmilesOps.cpp]: Unless read_text_to advances first...

**Comment by bp-kelley** [comment]
A bit mind bending, but looks ok.

**Comment by greglandrum** [comment]
Yeah, at some point it may be worth revisiting this code and writing a real parser

**Pull request merged**